userland/libtock/crt0.c

```diff
@@ -47,8 +47,9 @@ void _start(void* text_start,
             void* app_heap_break __attribute__((unused))) {
 
   // Allocate stack and data. `brk` to STACK_SIZE + got_size + data_size +
-  // bss_size from start of memory
-  uint32_t stacktop = (uint32_t)mem_start + STACK_SIZE;
+  // bss_size from start of memory. The stack must be aligned to a multiple of
+  // 4 so enforce that here.
```

Comment:
§ 5.2.1.2 requires that at a "public interface", the stack be 8-byte aligned. I think the initial stack location probably qualifies as "public" (though I'm not surprised the 4-byte works in practice, we should probably stick 8 on general principles)
http://infocenter.arm.com/help/topic/com.arm.doc.ihi0042f/IHI0042F_aapcs.pdf

I think it'd also be good to update this comment to indicate where the alignment requirement comes from